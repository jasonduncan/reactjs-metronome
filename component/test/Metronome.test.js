import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import Metronome from '../src/Metronome'

// mock the Worker class
class Worker { postMessage() { } }
window.Worker = Worker
window.URL.createObjectURL = jest.fn()

// mock the audio initialisation
jest.spyOn(Metronome.prototype, 'initAudio').mockImplementation(() => {})
const postMessage = jest.spyOn(window.Worker.prototype, 'postMessage')

// mount the component before each test
let wrapper
beforeEach(() => {
    wrapper = mount(<Metronome />)
})

// clear all the mocks after each test
afterEach(() => {
    jest.clearAllMocks()
})

// helper functions for clicking the UI buttons
const getBPM = () => wrapper.find('.metronome__bpm h1').text()
const clickPlus = () => wrapper.find('.slider__button--plus').simulate('click')
const clickMinus = () => wrapper.find('.slider__button--minus').simulate('click')
const clickPlayPause = () => wrapper.find('.metronome__play-pause').simulate('click')

it('should render Metronome correctly', () => {
    expect(renderer.create(<Metronome />).toJSON()).toMatchSnapshot()
})

it('should send starting interval to the worker', () => {
    expect(postMessage).toHaveBeenCalledTimes(1)
    expect(postMessage).toHaveBeenCalledWith({ message: "interval", interval: 25.0 })
})

it ('should increase BPM on + button click', () => {
    expect(getBPM()).toEqual('100')
    clickPlus()
    expect(getBPM()).toEqual('101')

    expect(renderer.create(<Metronome />).toJSON()).toMatchSnapshot()
})

it('should decrease BPM on - button click', () => {
    expect(getBPM()).toEqual('100')
    clickMinus()
    expect(getBPM()).toEqual('99')

    expect(renderer.create(<Metronome />).toJSON()).toMatchSnapshot()
})

it('should play on clicking play', () => {
    expect(wrapper.state('playing')).toEqual(false)
    clickPlayPause()
    expect(wrapper.state('playing')).toEqual(true)
    expect(postMessage).toHaveBeenLastCalledWith({ message: "start" })

    expect(renderer.create(<Metronome />).toJSON()).toMatchSnapshot()
})

it('should pause on clicking pause', () => {
    clickPlayPause()
    expect(wrapper.state('playing')).toEqual(true)

    clickPlayPause()
    expect(wrapper.state('playing')).toEqual(false)
    expect(postMessage).toHaveBeenLastCalledWith({ message: "stop" })

    expect(renderer.create(<Metronome />).toJSON()).toMatchSnapshot()
})

it('should clamp bpm to min value', () => {
    wrapper.setState((prevState) => ({ ...prevState, bpm: 41 }))
    clickMinus()
    expect(getBPM()).toEqual('40')
    clickMinus()
    expect(getBPM()).toEqual('40')

    expect(renderer.create(<Metronome />).toJSON()).toMatchSnapshot()
})

it('should clamp bpm to max value', () => {
    wrapper.setState((prevState) => ({ ...prevState, bpm: 199 }))
    clickPlus()
    expect(getBPM()).toEqual('200')
    clickPlus()
    expect(getBPM()).toEqual('200')

    expect(renderer.create(<Metronome />).toJSON()).toMatchSnapshot()
})

it('should post close to worker on unmount', () => {
    wrapper.unmount()
    expect(postMessage).toHaveBeenLastCalledWith({ message: "close" })
})