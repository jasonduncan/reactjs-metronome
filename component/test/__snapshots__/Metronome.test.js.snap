// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should clamp bpm to max value 1`] = `
<div
  className="metronome"
>
  <div
    className="metronome__top"
  >
    <div
      className="metronome__bpm"
    >
      <h1
        style={Object {}}
      >
        100
      </h1>
      <small
        style={Object {}}
      >
        BPM
      </small>
    </div>
    <div
      className="metronome__play-pause"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-play fa-w-14 "
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="metronome__slider"
    style={Object {}}
  >
    <div
      className="slider__button slider__button--minus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-minus fa-w-14 "
        data-icon="minus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "37.5%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={200}
        aria-valuemin={40}
        aria-valuenow={100}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "37.5%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
    <div
      className="slider__button slider__button--plus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`should clamp bpm to min value 1`] = `
<div
  className="metronome"
>
  <div
    className="metronome__top"
  >
    <div
      className="metronome__bpm"
    >
      <h1
        style={Object {}}
      >
        100
      </h1>
      <small
        style={Object {}}
      >
        BPM
      </small>
    </div>
    <div
      className="metronome__play-pause"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-play fa-w-14 "
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="metronome__slider"
    style={Object {}}
  >
    <div
      className="slider__button slider__button--minus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-minus fa-w-14 "
        data-icon="minus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "37.5%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={200}
        aria-valuemin={40}
        aria-valuenow={100}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "37.5%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
    <div
      className="slider__button slider__button--plus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`should decrease BPM on - button click 1`] = `
<div
  className="metronome"
>
  <div
    className="metronome__top"
  >
    <div
      className="metronome__bpm"
    >
      <h1
        style={Object {}}
      >
        100
      </h1>
      <small
        style={Object {}}
      >
        BPM
      </small>
    </div>
    <div
      className="metronome__play-pause"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-play fa-w-14 "
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="metronome__slider"
    style={Object {}}
  >
    <div
      className="slider__button slider__button--minus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-minus fa-w-14 "
        data-icon="minus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "37.5%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={200}
        aria-valuemin={40}
        aria-valuenow={100}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "37.5%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
    <div
      className="slider__button slider__button--plus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`should increase BPM on + button click 1`] = `
<div
  className="metronome"
>
  <div
    className="metronome__top"
  >
    <div
      className="metronome__bpm"
    >
      <h1
        style={Object {}}
      >
        100
      </h1>
      <small
        style={Object {}}
      >
        BPM
      </small>
    </div>
    <div
      className="metronome__play-pause"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-play fa-w-14 "
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="metronome__slider"
    style={Object {}}
  >
    <div
      className="slider__button slider__button--minus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-minus fa-w-14 "
        data-icon="minus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "37.5%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={200}
        aria-valuemin={40}
        aria-valuenow={100}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "37.5%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
    <div
      className="slider__button slider__button--plus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`should pause on clicking pause 1`] = `
<div
  className="metronome"
>
  <div
    className="metronome__top"
  >
    <div
      className="metronome__bpm"
    >
      <h1
        style={Object {}}
      >
        100
      </h1>
      <small
        style={Object {}}
      >
        BPM
      </small>
    </div>
    <div
      className="metronome__play-pause"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-play fa-w-14 "
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="metronome__slider"
    style={Object {}}
  >
    <div
      className="slider__button slider__button--minus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-minus fa-w-14 "
        data-icon="minus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "37.5%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={200}
        aria-valuemin={40}
        aria-valuenow={100}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "37.5%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
    <div
      className="slider__button slider__button--plus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`should play on clicking play 1`] = `
<div
  className="metronome"
>
  <div
    className="metronome__top"
  >
    <div
      className="metronome__bpm"
    >
      <h1
        style={Object {}}
      >
        100
      </h1>
      <small
        style={Object {}}
      >
        BPM
      </small>
    </div>
    <div
      className="metronome__play-pause"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-play fa-w-14 "
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="metronome__slider"
    style={Object {}}
  >
    <div
      className="slider__button slider__button--minus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-minus fa-w-14 "
        data-icon="minus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "37.5%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={200}
        aria-valuemin={40}
        aria-valuenow={100}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "37.5%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
    <div
      className="slider__button slider__button--plus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`should render Metronome correctly 1`] = `
<div
  className="metronome"
>
  <div
    className="metronome__top"
  >
    <div
      className="metronome__bpm"
    >
      <h1
        style={Object {}}
      >
        100
      </h1>
      <small
        style={Object {}}
      >
        BPM
      </small>
    </div>
    <div
      className="metronome__play-pause"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-play fa-w-14 "
        data-icon="play"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="metronome__slider"
    style={Object {}}
  >
    <div
      className="slider__button slider__button--minus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-minus fa-w-14 "
        data-icon="minus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h384c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
    <div
      className="rc-slider"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="rc-slider-rail"
        style={Object {}}
      />
      <div
        className="rc-slider-track"
        style={
          Object {
            "left": "0%",
            "right": "auto",
            "width": "37.5%",
          }
        }
      />
      <div
        className="rc-slider-step"
      />
      <div
        aria-disabled={false}
        aria-valuemax={200}
        aria-valuemin={40}
        aria-valuenow={100}
        className="rc-slider-handle"
        onBlur={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        role="slider"
        style={
          Object {
            "left": "37.5%",
            "right": "auto",
            "transform": "translateX(-50%)",
          }
        }
        tabIndex={0}
      />
      <div
        className="rc-slider-mark"
      />
    </div>
    <div
      className="slider__button slider__button--plus"
      onClick={[Function]}
      style={Object {}}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-plus fa-w-14 "
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
